boa_gc/src/lib.rs

```diff
@@ -367,12 +367,17 @@ impl Collector {
 
         // === Weak mark phase ===
         //
+        //
         // 1. Get the naive list of ephemerons that are supposedly dead or their key is dead and
         // trace all the ephemerons that have roots and their keys are live. Also remove from
         // this list the ephemerons that are marked but their value is dead.
         while let Some(eph) = weak.get() {
             // SAFETY: node must be valid as this phase cannot drop any node.
             let eph_ref = unsafe { eph.as_ref() };
+            let header = eph_ref.header();
+            if header.roots() > 0 {
```

Comment:
```suggestion
            if header.roots() != 0 {
```

It's an unsigned integer so `>` only excludes `0`, I think it would be clearer to just say `!= 0`